spring-social-facebook:1.0.0.RC1 dependency missing

It seems that a facebook dependency for org.springframework.social is breaking the build:

```
[ERROR] Failed to execute goal on project javascript-console-repo: Could not resolve dependencies for project de.fmaul:javascript-console-repo:amp:0.7-SNAPSHOT: Could not find artifact org.springframework.social:spring-social-facebook:jar:1.0.0.RC1 in alfresco-public (https://artifacts.alfresco.com/nexus/content/groups/public) -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :javascript-console-repo
```

A potential solution would be to remove org.springframework.social in the console-repo pom:
```
    <dependencies>
        <dependency>
            <groupId>${alfresco.groupId}</groupId>
            <artifactId>alfresco-repository</artifactId>
            <exclusions>
               <exclusion>
                  <groupId>org.springframework.social</groupId>
                  <artifactId>*</artifactId>
               </exclusion>
            </exclusions>
        </dependency>
    </dependencies>
```

[pom.xml.txt](https://github.com/share-extras/js-console/files/5341764/pom.xml.txt)

--- a/javascript-console-repo/pom.xml
+++ b/javascript-console-repo/pom.xml
@@ -82,7 +82,80 @@
     <dependencies>
         <dependency>
             <groupId>${alfresco.groupId}</groupId>
+            <artifactId>alfresco-data-model</artifactId>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.apache.chemistry.opencmis</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.apache.tika</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>commons-io</groupId>
+                    <artifactId>commons-io</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>${alfresco.groupId}</groupId>
             <artifactId>alfresco-repository</artifactId>
+            <exclusions>
+                <!-- trim down artifacts as best as possible (speed up build + avoid accidental dependencies to specific Alfresco versions) -->
+                <exclusion>
+                    <groupId>org.alfresco</groupId>
+                    <artifactId>alfresco-legacy-lucene</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.alfresco</groupId>
+                    <artifactId>alfresco-jlan-embed</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.alfresco</groupId>
+                    <artifactId>alfresco-mbeans</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.alfresco</groupId>
+                    <artifactId>alfresco-text-gen</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.alfresco.services</groupId>
+                    <artifactId>alfresco-events</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.alfresco</groupId>
+                    <artifactId>alfresco-greenmail</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.apache.chemistry.opencmis</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.apache.tika</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.apache.poi</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.springframework.social</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>commons-lang</groupId>
+                    <artifactId>commons-lang</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>commons-io</groupId>
+                    <artifactId>commons-io</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>com.google.guava</groupId>
+                    <artifactId>guava</artifactId>
+                </exclusion>
+            </exclusions>
         </dependency>
     </dependencies>
 

--- a/javascript-console-repo/src/main/amp/config/alfresco/module/de.fme.alfresco.JavascriptConsole-repo/module-context.xml
+++ b/javascript-console-repo/src/main/amp/config/alfresco/module/de.fme.alfresco.JavascriptConsole-repo/module-context.xml
@@ -21,7 +21,6 @@
         		<property name="renditionService" ref="RenditionService" />
         		<property name="ruleService" ref="RuleService" />
         		<property name="tagService" ref="TaggingService" />
-        		<property name="categoryService" ref="CategoryService" />
         		<property name="webDavService" ref="webdavService" />
         		<property name="auditService" ref="AuditService" />
         		<property name="sysAdminParams" ref="sysAdminParams" />

--- a/javascript-console-repo/src/main/java/de/fme/jsconsole/DumpService.java
+++ b/javascript-console-repo/src/main/java/de/fme/jsconsole/DumpService.java
@@ -1,6 +1,7 @@
 package de.fme.jsconsole;
 
 import java.io.Serializable;
+import java.math.BigInteger;
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Date;
@@ -13,7 +14,6 @@ import java.util.Map;
 import java.util.Set;
 import java.util.TreeMap;
 import java.util.concurrent.atomic.AtomicInteger;
-
 import org.alfresco.model.ContentModel;
 import org.alfresco.repo.admin.SysAdminParams;
 import org.alfresco.repo.jscript.ScriptNode;
@@ -34,7 +34,6 @@ import org.alfresco.service.cmr.repository.datatype.DefaultTypeConverter;
 import org.alfresco.service.cmr.repository.datatype.TypeConversionException;
 import org.alfresco.service.cmr.rule.Rule;
 import org.alfresco.service.cmr.rule.RuleService;
-import org.alfresco.service.cmr.search.CategoryService;
 import org.alfresco.service.cmr.security.AccessPermission;
 import org.alfresco.service.cmr.security.PermissionService;
 import org.alfresco.service.cmr.tagging.TaggingService;
@@ -46,16 +45,12 @@ import org.alfresco.service.cmr.workflow.WorkflowService;
 import org.alfresco.service.namespace.NamespaceException;
 import org.alfresco.service.namespace.NamespaceService;
 import org.alfresco.service.namespace.QName;
-import org.apache.commons.io.FileUtils;
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 import org.json.JSONArray;
 import org.json.JSONException;
 import org.json.JSONObject;
 import org.springframework.extensions.webscripts.ScriptValueConverter;
-
-import com.google.common.base.Function;
-import com.google.common.collect.Iterables;
 
 /**
  * Implements the 'jsconsole' Javascript extension object that is available in
@@ -87,8 +82,6 @@ public class DumpService {
 	private WorkflowService workflowService;
 
 	private RenditionService renditionService;
-
-	private CategoryService categoryService;
 
 	private TaggingService tagService;
 
@@ -227,12 +220,32 @@ public class DumpService {
 			if (contentReader != null) {
 				json.put("content encoding", contentReader.getEncoding());
 				json.put("content mimetype", contentReader.getMimetype());
-				json.put("content size", FileUtils.byteCountToDisplaySize(contentReader.getSize()));
+				json.put("content size", byteCountToDisplaySize(contentReader.getSize()));
 				json.put("content locale", contentReader.getLocale());
 				json.put("content lastModified", new Date(contentReader.getLastModified()));
 				json.put("content url", contentReader.getContentUrl());
 			}
 		}
+	}
+
+	// similar result to commons-io FileUtils#byteCountToDisplaySize(BigInteger)
+	// implemented here to avoid hard dependency (flagged by extension inspector)
+	private static String byteCountToDisplaySize(long size) {
+		String displaySize = String.valueOf(size) + " bytes";
+		if (size > 1024) {
+			BigInteger bis = BigInteger.valueOf(size);
+			BigInteger unitStep = BigInteger.valueOf(2^10);
+			BigInteger unitDivisor = BigInteger.valueOf(2^60);
+			String[] unitSuffixes = {" EB", " PB", " TB", " GB", " MB", " KB"};
+			for (String unitSuffix : unitSuffixes) {
+				if (bis.compareTo(unitDivisor) > 0) {
+					displaySize = String.valueOf(bis.divide(unitDivisor)) + unitSuffix;
+					break;
+				}
+				unitDivisor = unitDivisor.divide(unitStep);
+			}
+		}
+		return displaySize;
 	}
 
 	 /**
@@ -272,14 +285,12 @@ public class DumpService {
 		VersionHistory versionHistory = versionService.getVersionHistory(nodeRef);
 		if (versionHistory != null) {
 			json.put("version count", versionHistory.getAllVersions().size());
-			json.put("version count tooltip", Iterables.transform(versionHistory.getAllVersions(), new Function<Version, String>(){
-
-				@Override
-				public String apply(Version input) {
-					return input.getVersionProperties().toString();
-				}
-
-			}));
+			Collection<Version> allVersions = versionHistory.getAllVersions();
+			List<String> tooltipFragments = new ArrayList<>(allVersions.size());
+			for (Version version : allVersions) {
+				tooltipFragments.add(version.getVersionProperties().toString());
+			}
+			json.put("version count tooltip", tooltipFragments);
 		} else {
 			json.put("version count", "0");
 		}
@@ -525,10 +536,6 @@ public class DumpService {
 
 	public void setRenditionService(RenditionService renditionService) {
 		this.renditionService = renditionService;
-	}
-
-	public void setCategoryService(CategoryService categoryService) {
-		this.categoryService = categoryService;
 	}
 
 	public void setTagService(TaggingService tagService) {

--- a/javascript-console-repo/src/main/java/de/fme/jsconsole/ExecuteWebscript.java
+++ b/javascript-console-repo/src/main/java/de/fme/jsconsole/ExecuteWebscript.java
@@ -1,18 +1,20 @@
 package de.fme.jsconsole;
 
+import java.io.BufferedReader;
 import java.io.ByteArrayInputStream;
 import java.io.IOException;
 import java.io.InputStream;
+import java.io.InputStreamReader;
 import java.io.PrintWriter;
 import java.io.Reader;
 import java.io.StringReader;
 import java.io.StringWriter;
 import java.io.Writer;
 import java.nio.charset.Charset;
+import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
-
 import org.alfresco.repo.cache.SimpleCache;
 import org.alfresco.repo.content.MimetypeMap;
 import org.alfresco.repo.jscript.RhinoScriptProcessor;
@@ -26,8 +28,6 @@ import org.alfresco.service.cmr.repository.NodeRef;
 import org.alfresco.service.transaction.TransactionService;
 import org.alfresco.util.MD5;
 import org.alfresco.util.Pair;
-import org.apache.commons.io.IOUtils;
-import org.apache.commons.lang.StringUtils;
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 import org.json.JSONException;
@@ -225,9 +225,18 @@ public class ExecuteWebscript extends AbstractWebScript {
 	}
 
 	private String readScriptFromResource(String resource, boolean unwrapLines) throws IOException {
-        List<String> lines = IOUtils.readLines(getClass().getResourceAsStream(resource));
+	    BufferedReader reader = new BufferedReader(new InputStreamReader(getClass().getResourceAsStream(resource)));
+	    List<String> lines = new ArrayList<>(128);
+	    try {
+	      String line = null;
+	      while ((line = reader.readLine()) != null) {
+	        lines.add(line);
+	      }
+	    } finally {
+	      reader.close();
+	    }
 
-		StringBuffer script = new StringBuffer();
+		StringBuilder script = new StringBuilder();
 		for (String line : lines) {
 			if (unwrapLines) {
 				script.append(line.replace("\n", ""));
@@ -243,7 +252,7 @@ public class ExecuteWebscript extends AbstractWebScript {
 
 		LOG.debug("running script as user " + jsreq.runas);
 
-		if (StringUtils.isNotBlank(jsreq.runas)) {
+		if (jsreq.runas != null && !jsreq.runas.trim().isEmpty()) {
 			return AuthenticationUtil.runAs(new AuthenticationUtil.RunAsWork<JavascriptConsoleResult>() {
 				@Override
 				public JavascriptConsoleResult doWork() {
@@ -323,7 +332,7 @@ public class ExecuteWebscript extends AbstractWebScript {
 			JavascriptConsoleScriptObject javascriptConsole = printOutput == null ? new JavascriptConsoleScriptObject() : new JavascriptConsoleScriptObject(printOutput);
 			scriptModel.put("jsconsole", javascriptConsole);
 
-			if (StringUtils.isNotBlank(spaceNodeRef)) {
+			if (spaceNodeRef != null && !spaceNodeRef.trim().isEmpty()) {
 				javascriptConsole.setSpace(this.scriptUtils.getNodeFromString(spaceNodeRef));
 			} else {
 				Object ch = scriptModel.get("companyhome");
@@ -336,7 +345,7 @@ public class ExecuteWebscript extends AbstractWebScript {
 			scriptModel.put("space", javascriptConsole.getSpace());
 
 			ScriptNode documentNode = null;
-			if (StringUtils.isNotBlank(documentNodeRef)) {
+			if (documentNodeRef != null && !documentNodeRef.trim().isEmpty()) {
 				documentNode = this.scriptUtils.getNodeFromString(documentNodeRef);
 				scriptModel.put("document", documentNode);
 			}
@@ -383,7 +392,7 @@ public class ExecuteWebscript extends AbstractWebScript {
 					res.setHeader(WebScriptResponse.HEADER_LOCATION, location);
 				}
 
-				if (StringUtils.isNotBlank(template)) {
+				if (template != null && !template.trim().isEmpty()) {
 					PerfLog freemarkerPerf = new PerfLog(LOG).start();
 					TemplateProcessor templateProcessor = getContainer().getTemplateProcessorRegistry()
 							.getTemplateProcessorByExtension("ftl");
@@ -424,7 +433,7 @@ public class ExecuteWebscript extends AbstractWebScript {
 	 * @param templateModel
 	 *            template model
 	 */
-	final private void mergeScriptModelIntoTemplateModel(ScriptContent scriptContent, Map<String, Object> scriptModel,
+	private final void mergeScriptModelIntoTemplateModel(ScriptContent scriptContent, Map<String, Object> scriptModel,
 			Map<String, Object> templateModel) {
 		// determine script processor
 		ScriptProcessor scriptProcessor = getContainer().getScriptProcessorRegistry().getScriptProcessor(scriptContent);
@@ -448,7 +457,7 @@ public class ExecuteWebscript extends AbstractWebScript {
 	 * @param writer
 	 *            where to output
 	 */
-	final protected void renderFormatTemplate(String format, Map<String, Object> model, Writer writer) {
+	protected final void renderFormatTemplate(String format, Map<String, Object> model, Writer writer) {
 		format = (format == null) ? "" : format;
 
 		String templatePath = getDescription().getId() + "." + format;

--- a/javascript-console-repo/src/main/java/de/fme/jsconsole/PerfLog.java
+++ b/javascript-console-repo/src/main/java/de/fme/jsconsole/PerfLog.java
@@ -5,7 +5,6 @@ package de.fme.jsconsole;
 
 import java.text.MessageFormat;
 
-import org.apache.commons.lang.StringUtils;
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 
@@ -20,7 +19,7 @@ import org.apache.commons.logging.LogFactory;
 public class PerfLog {
 
 	private static final int DEFAULT_ASSERT_PERFORMANCE = 2000;
-	private Log LOG = LogFactory.getLog(PerfLog.class);
+	private Log logger = LogFactory.getLog(PerfLog.class);
 	private long startTime;
 
 	/**
@@ -30,7 +29,7 @@ public class PerfLog {
 	 */
 	public PerfLog(Log logger) {
 		if (logger != null) {
-			LOG = logger;
+			this.logger = logger;
 		}
 	}
 
@@ -49,10 +48,8 @@ public class PerfLog {
 	 */
 	public PerfLog start(String message, Object... params) {
 		startTime = System.currentTimeMillis();
-		if (LOG.isInfoEnabled() || LOG.isWarnEnabled()) {
-			if (StringUtils.isNotEmpty(message)) {
-				LOG.info(MessageFormat.format(message, params));
-			}
+		if (logger.isInfoEnabled() && message != null && !message.trim().isEmpty()) {
+			logger.info(MessageFormat.format(message, params));
 		}
 		return this;
 	}
@@ -73,13 +70,11 @@ public class PerfLog {
 	public long stop(int assertPerformanceOf, String message, Object... params) {
 		long endTime = System.currentTimeMillis();
 		long neededTime = endTime - startTime;
-		if (LOG.isInfoEnabled() || LOG.isWarnEnabled()) {
-			if (neededTime < assertPerformanceOf) {
-				LOG.info("(OK) " + neededTime + " ms:" + MessageFormat.format(message, params));
-			} else {
-				LOG.warn("(WARNING) " + neededTime + " ms: " + MessageFormat.format(message, params));
-			}
-
+		boolean inTime = neededTime < assertPerformanceOf;
+        if (logger.isInfoEnabled() && inTime) {
+			logger.info("(OK) " + neededTime + " ms:" + MessageFormat.format(message, params));
+		} else if (logger.isWarnEnabled() && !inTime){
+			logger.warn("(WARNING) " + neededTime + " ms: " + MessageFormat.format(message, params));
 		}
 		return neededTime;
 

